src/server/routes/register.ts

```diff
@@ -0,0 +1,76 @@
+import { Request, Router } from 'express';
+import { authenticate } from '@/server/lib/idapi/auth';
+import { create } from '@/server/lib/idapi/user';
+import { logger } from '@/server/lib/logger';
+import { renderer } from '@/server/lib/renderer';
+import { Routes } from '@/shared/model/Routes';
+import { ResponseWithRequestState } from '@/server/models/Express';
+import { trackMetric } from '@/server/lib/AWS';
+import { Metrics } from '@/server/models/Metrics';
+import { PageTitle } from '@/shared/model/PageTitle';
+import { handleAsyncErrors } from '@/server/lib/expressWrappers';
+import { setIDAPICookies } from '@/server/lib/setIDAPICookies';
+import { getConfiguration } from '../lib/getConfiguration';
+
+const router = Router();
+
+router.get(
+  Routes.REGISTRATION,
+  (req: Request, res: ResponseWithRequestState) => {
+    const html = renderer(Routes.REGISTRATION, {
+      requestState: res.locals,
+      pageTitle: PageTitle.REGISTRATION,
+    });
+    res.type('html').send(html);
+  },
+);
+
+router.post(
+  Routes.REGISTRATION,
+  handleAsyncErrors(async (req: Request, res: ResponseWithRequestState) => {
+    let state = res.locals;
+
+    const { email = '' } = req.body;
+    const { password = '' } = req.body;
+
+    const { returnUrl } = state.pageData;
```

Comment:
```suggestion
    const { returnUrl } = state.queryParams;
```
`state.queryParams.returnUrl` is typed so that it always exists, so you don't need the call to `getConfiguration` to get the `defaultReturnUri`